### ISSUE TITLE
javax.ws.rs.core.Response streams re-encoded into UTF-8

### ISSUE THREAD

**Issue by Lucas3oo** [spring-projects/spring-boot#14616]
I am using Jersey in Spring boot.
It seems that Spring in re-encoding the HTTP response in to UTF-8 even though I am specifying UTF-16LE.

I am using io.spring.platform:platform-bom:Cairo-SR3
which uses Jersey 2.26.

In a Jetty/Jersey 2.26 minimal setup it works as it should.

What I want to generate is actually a CSV file with the UTF-16LE BOM in the beginning. But the server always returns UTF-8 and there are no BOM

So it seems that some kind of Spring wrapper around Jersey re-encodes the response streams.

Typical code:

```java 
@Path("/csv")
public class HelloResource {
  @GET
  @Produces("application/csv;charset=UTF-16")
  public Response getCsvFile() {
   StreamingOutput streamingOutput = new  StreamingOutput() {
      public void write(OutputStream output) throws IOException, WebApplicationException {
        OutputStreamWriter writer = new OutputStreamWriter(output,
            StandardCharsets.UTF_16LE.newEncoder());
        writer.write('\ufeff'); // since UTF_16LE is used we need to "reverse" the BOM, the BOM for UTF_16LE is actually FFFE
        writer.write("Ärkel");
        writer.flush();    
      }
    };
    return Response.ok(streamingOutput)
        .header("Content-Disposition", String.format("attachment; filename=\"%s\"", "some.csv"))
        .build();
  }
}
```

**Comment by wilkinsona** [comment]
It could be the character encoding filter that will default encoding to UTF-8. However, it’s very hard to say with any certainty based on a small snippet of code. You may not even have the filter on the classpath. Can you please provide a minimal sample (something we can `git clone` or unzip) that reproduces the behaviour you have described?

**Comment by Lucas3oo** [comment]
Hi Andy

Yeah, I can setup two small project on GitHub,

I did check all the filters that where configured, and as you said it has the character encoding filter. But when I study that code it seems to only add the character encoding header to the response (and request). It will not do anything with the response stream.

I actually set the sprig properties to force character encoding to UTF-16 but it is only for the header , not the body/content.

Cheers
Lucas
